tgui/packages/tgui/interfaces/NtosNotepad.tsx

```diff
@@ -1,39 +1,511 @@
+/**
+ * @file
+ * @copyright 2022 raffclar
+ * @license MIT
+ */
+
 import { NtosWindow } from '../layouts';
-import { useBackend } from '../backend';
-import { Stack, Input, Section } from '../components';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Section, TextArea, MenuBar, Divider } from '../components';
+import { Component, createRef, RefObject } from 'inferno';
+import { createLogger } from '../logging';
+import { Dialog, UnsavedChangesDialog } from '../components/Dialog';
+
+const logger = createLogger('NtosNotepad');
+
+const DEFAULT_DOCUMENT_NAME = 'Untitled';
+
+type PartiallyUnderlinedProps = {
+  str: string;
+  indexStart: number;
+};
+
+const PartiallyUnderlined = (props: PartiallyUnderlinedProps) => {
+  const { str, indexStart } = props;
+  const start = str.substring(0, indexStart);
+  const underlined = str.substring(indexStart, indexStart + 1);
+  const end = indexStart < str.length - 1 ? str.substring(indexStart + 1) : '';
+  return (
+    <>
+      {start}
+      <span style={{ 'text-decoration': 'underline' }}>{underlined}</span>
+      {end}
+    </>
+  );
+};
+
+enum Dialogs {
+  NONE = 0,
+  UNSAVED_CHANGES = 1,
+  OPEN = 2,
+  ABOUT = 3,
+}
+
+type MenuBarProps = {
+  save: () => void;
+  exit: () => void;
+  newNote: () => void;
+  cutSelected: () => void;
+  copySelected: () => void;
+  pasteClipboard: () => void;
+  deleteSelected: () => void;
+  showStatusBar: boolean;
+  setShowStatusBar: (boolean) => void;
+  wordWrap: boolean;
+  setWordWrap: (boolean) => void;
+  aboutNotepadDialog: () => void;
```

Comment:
These callbacks should have an `on*` prefix, but it's not components code, so it's not too critical.